core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1355,6 +1402,60 @@ class KafkaApis(val requestChannel: RequestChannel,
       ))
   }
 
+  def handleDescribeTopicsRequest(request: RequestChannel.Request): Unit = {
+    val describeTopicsRequest = request.body[DescribeTopicsRequest]
+
+    val topics = scala.collection.mutable.Map[String, Int]()
+    describeTopicsRequest.data.topics.forEach { topic =>
+      if (topic.name == null || topic.firstPartitionId() < 0) {
+        throw new InvalidRequestException(s"Topic name and first partition id must be set.")
+      }
+      topics.put(topic.name(), topic.firstPartitionId())
+    }
+
+    val fetchAllTopics = topics.isEmpty
```

Comment:
How does `firstPartitionIndex` work when requesting all topics? Will we still return all topics in the initial response?

Reply:
If there is no topic in the request, it means the client wants all the topics.
firstPartitionIndex is a topic level argument. It only works in the case where topics are provided.

Reply:
Right. My question is how the user can iterate all topics in the presence of the limit?